aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutboxTest.java

```diff
@@ -294,7 +294,7 @@ public void existingUpdateIncomingCreationYieldsError() throws DataStoreExceptio
             .name("Tony with improvements applied")
             .build();
         PendingMutation<BlogOwner> existingUpdate =
-            PendingMutation.update(modelInExistingMutation, BlogOwner.class);
+            PendingMutation.update(modelInExistingMutation, BlogOwner.class, null);
```

Comment:
this is repeat often it seems, maybe an overload is worth it?

Reply:
^ Agree.

For these predicates, around the entire code base, we either need to:

1. Provide overloads. One does not require a predicate, the other requires a _non-null_ predicate. This makes it clear to the user what's going on, and we refuse to have our user supply a semantic `null`.

2. Don't provide an overload, just have a required param. If we do this, we need  a factory which can create a non-null sentinel, e.g.

```
QueryPredicates.matchAll() // returns something that is not `null`.
```

Then internally, we can do like:
```
if (isSentinelValue(queryPredicate)) {
    // do whatever the null meant.
}
```

in practice, that is:
```
if QueryPredicates.matchAll().equals(queryPredicate)) {
    // match all!
}
```

Reply:
I addressed this by adding overloads for now.